presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/SemanticException.java

```diff
@@ -16,39 +16,51 @@
 import com.facebook.presto.sql.tree.Node;
 import com.facebook.presto.sql.tree.NodeLocation;
 
+import java.util.Optional;
+
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class SemanticException
         extends RuntimeException
 {
     private final SemanticErrorCode code;
-    private final Node node;
+    private final Optional<NodeLocation> location;
+
+    public SemanticException(SemanticErrorCode code, String format, Object... args)
+    {
+        this(code, Optional.empty(), format, args);
+    }
 
     public SemanticException(SemanticErrorCode code, Node node, String format, Object... args)
     {
-        super(formatMessage(format, node, args));
+        this(code, node == null ? Optional.empty() : node.getLocation(), format, args);
+    }
+
+    public SemanticException(SemanticErrorCode code, Optional<NodeLocation> location, String format, Object... args)
+    {
+        super(formatMessage(format, location, args));
```

Comment:
is location guaranteed to be non-null here?
do we need a requireNonNull for `location`

Reply:
Location is not null in the node class.